src/browser/api/external_window.ts

```diff
@@ -39,18 +44,44 @@ export function closeExternalWindow(identity: Identity): void {
   externalWindow.forceExternalWindowClose();
 }
 
-export async function disableExternalWindowUserMovement(identity: Identity): Promise<void> {
-  const externalWindow = getExternalWindow(identity);
+export async function disableExternalWindowUserMovement(target: Identity, requestor?: Identity): Promise<void> {
+  const externalWindow = getExternalWindow(target);
   const injectionBus = getInjectionBus(externalWindow);
   await injectionBus.set({ userMovement: false });
   externalWindow.emit('user-movement-disabled');
+
+  // Register a subscription that will fire every time requesting identity closes
+  // and if all requestors for disabling frame are gone - enable the frame.
+  if (requestor) {
+    const key = getKey(externalWindow);
+    const subscriptionKey = `disable-external-window-user-movement-${key}`;
+    const onRequestorClose = async () => {
+      let requestorCount = disabledUserMovementRequestorCount.get(key) || 0;
```

Comment:
requestorCount scares me... wouldn't it be more foolproof to push\pop requestors to/from an array (instead of a map) and just check to see if the array is empty onRequestorClose?

Reply:
We do need a map to do the lookup to check which window has requestors. Are you saying you'd prefer to have an array of requestors for a particular window, instead of an integer (count)?

Reply:
yes. it's a matter of preference, so definitely ignore this if you disagree. in my experience, it's easy to get these counters out of sync (assuming other people will poke this code in the future), in oppose to a more deterministic list of identities. this would also remove the need to decrement from enableExternaWindowUserMovement.
I'd would've used Map<string, Identity[]>